enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,279 @@
+
+#### What kinds of roles exist related to managedClusterSets and managedClusters
+In General, there are three kinds of roles related to managedClusterSet.
+1. Cluster admin
+   - Cluster admin could `create`, `get`, `list`, `update`, `delete` all managedClusterSets.
+   - Cluster admin could `bind` all managedClusterSets to all namespaces.
+   - Cluster admin could `create`, `get`, `list`, `update`, `delete` all managedClusters.
+   - Only Cluster admin could `accept` managedClusters.
+   
+2. Controller/agent which run in managedClusters
+   - The Controller/agent should not `create`, `get`, `list`, `update`, `delete` managedClusterSets.
+   - The Controller/agent should not `bind` managedClusterSet.
+   - The Controller/agent only should `get` related managedCluster and `update` the managedCluster status and labels.
+
+3. Non Cluster admin users
+   - These users should `get` some managedClusterSets, and could not `create`, `update`, `delete`, `list` managedClusterSets.
+   - These users should `bind` some managedClusterSets to their namespaces.
+   - These users may need to `create` managedClusters(For Resource Group Purpose managedClusterSet[d], in each group, the group members may need to create the managedClusters). They could also `delete` these managedClusters, and `update` these managedClusters labels.
+
+#### Who can join a managedCluster to a managedClusterSet
+In this proposal, managedClusterSet could use any labels to select managedClusters. If one user adds a label to a managedClusters, it may lead to this managedClusters added to a managedClusterSet. So this question also means "who can add/update a label for managedClusters".
+
+1. Cluster admin
+Cluster admin could add any labels to any managedClusters.
+
+2. Controller/agent which run in managedClusters
+The controller/agent may add labels like: `region: apac`, `cloud: Amazon`, `middlewareEnabled: true`
+These labels should not be changed by the user. And these labels mean these clusters have some attributes.
+- `region: apac` means the cluster is provisioned in apac region
+- `cloud: Amazon` means the cluster is provisioned in Amazon cloud provider
+- `middlewareEnabled: true` menas this cluster enables the middleware.
+
+3. Non Cluster admin users
+Currently, non cluster admin users may `add/update` some managedClusters labels. And these labels may lead to the managedClusters added to a managedClusterSet.
+
+But for some managedClusterSet(like: Disaster Recovery ManagedClusterSet[a], Resource Group ManagedClusterSet[d]), we do not want any users to add their managedClusters to these managedClusterSets. 
+
+For these managedClusterSets, we need a way to identify them. 
+We popose use the prefix `cluster.open-cluster-management.io/` in managedClusterSet `spec.clusterLabel.key` to identify the managedClusterSet which need "special" permission to join. (This prefix may be configured. But in this proposal, it is const)
```

Comment:
This is a good idea. We can add restrictions to labels in our domain, because we own that area.